### ISSUE TITLE
BWA-MEM vs BWA-MEM2.X reproducibility

### ISSUE THREAD

**Issue by tyamaguchi-ucla** [bwa-mem2/bwa-mem2#113]
Hi,

We've conducted some benchmarking in order to compare BWA-MEM to BWA-MEM2.x using multi-library samples (reference  genome - GRCh38).

It looks like BWA-MEM2 and 2.1 produced the almost identical results but BWA-MEM2.X treated non-primary reads a bit differently than BWA-MEM 0.7.17-r1188?

The stats were generated using `samtools stats` (v 1.11 using htslib 1.11)

stats | BWA-MEM2 / BWA-MEMv2.1 | BWA-MEM / BWA-MEMv2.1
------- | -------- | ---------
raw total sequences:             |        1.0000000  |   1.0000000
filtered sequences:                 |           NaN       |    NaN
sequences:         |                      1.0000000  |   1.0000000
is sorted:                   |            1.0000000    | 1.0000000
1st fragments:            |               1.0000000    | 1.0000000
last fragments:            |              1.0000000    | 1.0000000
reads mapped:             |               1.0000000   |  0.9999863
reads mapped and paired: |                1.0000001 |    0.9999732
reads unmapped:                 |         0.9999950  |   1.0020125
reads properly paired:           |        0.9999999|     0.9966227
reads paired:                           | 1.0000000     |1.0000000
reads duplicated:                     |   0.9999949  |   1.0113399
reads MQ0:                              | 1.0000009    | **0.7895199**
reads QC failed:                        |       NaN          | NaN
non-primary alignments:           |       0.9999757|     **2.3400824**
total length:                      |      1.0000000     |1.0000000
bases mapped:                 |           1.0000000    | 0.9999873
bases mapped (cigar):      |              1.0000000|     1.0042651
bases trimmed:                   |              NaN        |   NaN
bases duplicated:                 |       0.9999958|     1.0111430
mismatches:                           |   0.9999987 |    1.0571385
error rate:                             | 0.9999987    | 1.0526487
average length:                     |     1.0000000     |1.0000000
maximum length:                   |       1.0000000   |  1.0000000
average quality:                      |   1.0000000     |1.0000000
insert size average:                 |    1.0000000    | 0.9990121
insert size standard deviation:|          1.0000000 |    1.0015860
inward oriented pairs:               |    0.9999991     |0.9965072
outward oriented pairs:              |    1.0000477    | **1.1132505**
pairs with other orientation:        |    0.9999726     |**1.2644096**
pairs on different chromosomes: |         1.0000416 |    **1.3031953**

We used the following command for alignment then used picard to sort and mark duplicates. Also we used slightly different numbers of threads but bwa-mem2 vs 2.1 gave the (almost) identical outputs.

bwa-mem(2)       mem       -t 50~64       -M       -R  (read-group info)  genome.fa       R1.fastq.gz       R2.fastq.gz |    samtools
      view       -@ 50       -S       -b >       output.bam

Any thoughts on the differences? Please let me know if you need more information.

Thanks,

Taka Yamaguchi

**Comment by yuk12** [comment]
Could you try with the latest commit and check?

**Comment by zhuchcn** [comment]
Hi, thanks for the effort of impoving the program! Are you referring to the latest commit on master branch?

**Comment by yuk12** [comment]
Yes, the master branch. If even after that you see the differences then can you provide a small subset of reads to reproduce and fix it?

**Comment by tyamaguchi-ucla** [comment]
Hi,

We just tried the latest commit.

For outward oriented pairs, the value changed from 1.1132505 to 0.9778192, which is closer to BWA-MEM 0.7.17-r1188 but otherwise the other stats remained the same.

Do you guys have access to the PCAWG datasets or https://ega-archive.org/studies/EGAS00001000900  by any chance?
